### ISSUE TITLE
[Release] Missing Ubuntu packages when multiple patch releases have been done

### ISSUE THREAD

**Issue by raulcd** [apache/arrow#39725]
### Describe the bug, including details regarding any error messages, version, and platform.

As raised on this Mailing list thread:
https://lists.apache.org/thread/1jcx51r077mzlt8yj55k05t6md6ybtqf

There seems to be an issue on the released Ubuntu packages when we have done more than one patch release. 
I think this was the first time we did more than one patch release for a version. I just released 15.0.0 and tried to re-release 14.0.2 for Ubuntu but I can't see the package for 14.0.2 .Example for libgandiva:
```
libgandiva-glib800/jammy 8.0.0-1 amd64
  Gandiva is a toolset for compiling and evaluating expressions

libgandiva-glib900/jammy 9.0.0-1 amd64
  Gandiva is a toolset for compiling and evaluating expressions

libgandiva1000/jammy 10.0.1-1 amd64
  Gandiva is a toolset for compiling and evaluating expressions

libgandiva1100/jammy 11.0.0-1 amd64
  Gandiva is a toolset for compiling and evaluating expressions

libgandiva1200/jammy 12.0.1-1 amd64
  Gandiva is a toolset for compiling and evaluating expressions

libgandiva1300/jammy 13.0.0-1 amd64
  Gandiva is a toolset for compiling and evaluating expressions

libgandiva1400/jammy 14.0.1-1 amd64
  Gandiva is a toolset for compiling and evaluating expressions

libgandiva1500/jammy 15.0.0-1 amd64
  Gandiva is a toolset for compiling and evaluating expressions
```

### Component(s)

Release

**Comment by kou** [comment]
(I'll write details tomorrow.)

**Comment by kou** [comment]
Here are our APT repository update steps:

1. RC: Download the current APT repository metadata (`dists/` directory has them)
2. RC: Generate APT repository metadata from newly added `.deb`s (We just generate APT repository metadata for 14.0.2 `.deb`s when we release 14.0.2)
3. RC: Merge 1. and 2.
4. RC: Upload 3. and newly added `.deb`s to `ubuntu-rc/`. They are recorded to `uploaded-files.txt`: https://apache.jfrog.io/artifactory/arrow/ubuntu-rc/uploaded-files.txt
5. Release: Copy 4. to `ubuntu/` based on `uploaded-files.txt`

I retried 1.-4. when we release 14.0.2 because I forgot to include Ubuntu 23.04 at the first try. (I just do for Ubuntu 20.04 and 22.04 at the first try.)
I just did 1.-4. only for Ubuntu 23.04 because I already done them for Ubuntu 20.04 and 22.04. This is the problem.

The `uploaded-files.txt` included files only for Ubuntu 23.04 because my second try targeted only Ubuntu 23.04.

So, 5. just copied 14.0.2 files only for Ubuntu 23.04. 14.0.2 files for Ubuntu 20.04 and 22.04 exist in `ubuntu-rc/` but weren't copied.
We can copy `.deb`s for 14.0.2 but can't copy APT repository metadata because the latter is already overwritten. We need to regenerate APT repository metadata from all existing `.deb`s. We can do it but it's time consuming and we have 15.0.0 now. So we may abandon 14.0.2 for Ubuntu 20.04 and 22.04.
